doc/example_query_logger.md

```diff
@@ -0,0 +1,90 @@
+# Query Logger Example
+
+This short guide will demonstrate how to setup ecAudit as a pure Query Logger in combination with the [Chronicle Logger](chronicle_logger.md) backend.
+In this example all CQL queries will be logged without any filtering, and users will be able to connect without any credentials.
+
+## cassandra.yaml
+
+Change the following setting in your ```cassandra.yaml```.
+
+```
+role_manager: com.ericsson.bss.cassandra.ecaudit.auth.AuditRoleManager
+```
+
+Leave the ```authenticator``` and ```authorizer``` settings at their default.
+
+```
+authenticator: AllowAllAuthenticator
+authorizer: AllowAllAuthorizer
+```
+
+
+## cassandra-env.sh
+
+Add the following JVM option to the ```cassandra-env.sh``` or the ```cassandra.in.sh```.
+
+```
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -Dcassandra.custom_query_handler_class=com.ericsson.bss.cassandra.ecaudit.handler.AuditQueryHandler"
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -Decaudit.filter_type=NONE"
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -da:net.openhft..."
+```
+
+**Note:** If these settings configured in the ```cassandra-env.sh```,
+consider that the ```JVM_EXTRA_OPTS``` variable is consumed at the end of the file,
+so make sure to add the lines *before* they are consumed.
+
+In other words, the last lines of the ```cassandra-env.sh``` should look something like this:
+
+```
+...
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -Dcassandra.custom_query_handler_class=com.ericsson.bss.cassandra.ecaudit.handler.AuditQueryHandler"
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -Decaudit.filter_type=NONE"
+JVM_EXTRA_OPTS="$JVM_EXTRA_OPTS -da:net.openhft..."
+JVM_OPTS="$JVM_OPTS $JVM_EXTRA_OPTS"
+```
+
+
+## audit.yaml
+
+Setup the ```audit.yalm``` as follows to roll a new Chronicle log file every minute,
+and store only fields which are relevant in this example.
+Also, use ```post_logging``` to make sure there is one audit record per CQL request.
+
+```
+log_timing_strategy: post_logging
+
+logger_backend:
+    - class_name: com.ericsson.bss.cassandra.ecaudit.logger.ChronicleAuditLogger
+      parameters:
+      - log_dir: /var/lib/cassandra/audit
+        roll_cycle: MINUTELY
+        fields: TIMESTAMP, OPERATION, BATCH_ID, STATUS
+```
+
+## eclog
+
+The query logs contain binary data since they are written with the Chronicle logger backend.
+
+The ```eclog``` tool will print the content in a human readable format.
+With a ```eclog.yaml``` file it is possible to get a custom format of each record.
+For example.
+
+```
+log_format: "${TIMESTAMP},{?${BATCH_ID}?},${STATUS},${OPERATION}"
```

Comment:
maybe "{?${BATCH_ID}?}," should be "{?${BATCH_ID},?}" so that we don't get an "empty field" (",,") in the example below...
Or if we want to make the example even simpler - just use the default format -> and not have the eclog.yaml in the example...

Reply:
Intention with this was to produce a typical csv format. In such a case I believe you do want the "empty field" to simplify parsing. Right? So for this reason the default format isn't what I want.

On the other hand, using comma ',' as separator isn't a great idea since it is used a lot in OPERATION. So, could make sense to stick with the pipe '|' as separator.

Since this is a small, but full, example on how to use and setup ecAudit I thought this was a good opportunity to demonstrate the nice formatting features of the eclog tool. As a nice bonus we'll get pretty dates in the example output below instead of ms since epoch.

Reply:
True - if we this the output to have csv format - then we should keep the "extra" separator.